shims/spark300/src/main/scala/com/nvidia/spark/rapids/shims/spark300/Spark300Shims.scala

```diff
@@ -255,8 +255,9 @@ class Spark300Shims extends SparkShims {
                 }
               case _ =>
             }
-            if (ZoneId.of(a.timeZoneId.get).normalized() != GpuOverrides.UTC_TIMEZONE_ID) {
-              willNotWorkOnGpu("Only UTC zone id is supported")
+            a.timeZoneId.foreach {
+              case zoneId if ZoneId.of(zoneId).normalized() != GpuOverrides.UTC_TIMEZONE_ID =>
+                willNotWorkOnGpu(s"Only UTC zone id is supported. Actual zone id: $zoneId")
```

Comment:
Isn't this going to trigger a `MatchError` if this one case doesn't match?  e.g.:
```
scala> Some(2).foreach{ case x if x != 2 => println(x) }
scala.MatchError: 2 (of class java.lang.Integer)
  at $anonfun$res1$1(<console>:26)
  at scala.runtime.java8.JFunction1$mcVI$sp.apply(JFunction1$mcVI$sp.java:23)
  at scala.collection.immutable.List.foreach(List.scala:392)
  ... 47 elided
```

Reply:
good catch, it's either should be a `collect` or we need to have a default `case _`

Reply:
Or let's not out-fancy ourselves and just do it in a very straightforward way:
```scala
a.timeZoneId.foreach { zoneId =>
  if (ZoneId.of(zoneId).normalized() != GpuOverrides.UTC_TIMEZONE_ID) {
    willNotWorkOnGpu(s"Only UTC zone id is supported. Actual zone id: $zoneId")
  }
}
```

Reply:
this will do too

Reply:
And we should factor it out since this fragment is used many times

Reply:
Thanks for the suggestions. Will update the PR.